nltools/data/brain_data.py

```diff
@@ -1205,26 +1205,17 @@ def apply_mask(self, mask):
             masked: (Brain_Data) masked Brain_Data object
 
         """
-
-        if isinstance(mask, Brain_Data):
-            mask = mask.to_nifti()  # convert to nibabel
-
-        if not isinstance(mask, nib.Nifti1Image):
-            if isinstance(mask, six.string_types):
-                if os.path.isfile(mask):
-                    mask = nib.load(mask)
-                if not ((self.mask.get_affine() == mask.get_affine()).all()) & (self.mask.shape[0:3] == mask.shape[0:3]):
-                    mask = resample_img(mask, target_affine=self.mask.get_affine(), target_shape=self.mask.shape)
-            else:
-                raise ValueError("Mask is not a nibabel instance, Brain_Data "
-                                 "instance, or a valid file name.")
-
+        mask = check_brain_data(mask)
         masked = deepcopy(self)
-        nifti_masker = NiftiMasker(mask_img=mask)
-        masked.data = nifti_masker.fit_transform(self.to_nifti())
-        masked.nifti_masker = nifti_masker
-        if (len(masked.shape()) > 1) & (masked.shape()[0] == 1):
-            masked.data = masked.data.flatten()
+
+        if masked.shape()[0] == mask.shape()[0]:
```

Comment:
I don't think this solution is adequate because line 1208 above (`mask = check_brain_data(mask)`) doesn't explicitly check for what _type_ of data is within the mask. We should either ensure that the mask contains bools only, or change this solution to `masked.data[~mask.data.astype(bool)] = 0`. In fact recent version of numpy will fail with a `TypeError` if you try to invert an array that contains floats using `~`.

Reply:
Changed the implementation, but now force cast as boolean as suggested.